cmd/hack.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+  "log"
+
+  "github.com/Originate/gt/cmd/config"
+  "github.com/Originate/gt/cmd/git"
+  "github.com/Originate/gt/cmd/script"
+  "github.com/Originate/gt/cmd/steps"
+  "github.com/Originate/gt/cmd/util"
+
+  "github.com/spf13/cobra"
+)
+
+var abortFlag bool
+var continueFlag bool
+
+var hackCmd = &cobra.Command{
+  Use:   "hack",
+  Short: "Create a new feature branch off the main development branch",
+  Long:  `Create a new feature branch off the main development branch`,
+  Run: func(cmd *cobra.Command, args []string) {
+    steps.Run(steps.RunOptions{
+      Command: "hack",
+      IsAbort: abortFlag,
+      IsContinue: continueFlag,
+      IsSkip: false,
+      SkipMessage: "",
+      StepListGenerator: func() steps.StepList {
+        targetBranchName := checkPreconditions(args)
+        return getStepList(targetBranchName)
+      },
+    })
+  },
+}
+
+func checkPreconditions(args []string) string {
+  if len(args) == 0 {
+    util.ExitWithErrorMessage("No branch name provided.")
+  }
+  targetBranchName := args[0]
+  if config.HasRemoteOrigin() {
+    fetchCmd := []string{"git", "fetch", "--prune"}
+    fetchErr := script.RunCommand(fetchCmd)
+    if fetchErr != nil {
+      log.Fatal(fetchErr)
+    }
+  }
+  git.EnsureDoesNotHaveBranch(targetBranchName)
+  return targetBranchName
+}
+
+func getStepList(targetBranchName string) steps.StepList {
+  mainBranchName := config.GetMainBranch()
+  stepList := steps.StepList{}
```

Comment:
Not possible at the moment as I am not actually returning this variable. It is passed to `steps.Wrap` which returns a new stepList.